src/pkg/utils/image.go

```diff
@@ -7,15 +7,32 @@ package utils
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/google/go-containerregistry/pkg/crane"
 	"os"
 	"path/filepath"
+	"regexp"
 
 	"github.com/defenseunicorns/zarf/src/pkg/transform"
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/layout"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+// GetImageWithSha returns the image reference and the sha256 digest of the image.
+func GetImageWithSha(imgSrc string) (string, error) {
+	regex := regexp.MustCompile(`^.*@sha256:[a-f0-9]{64}$`)
+	if regex.MatchString(imgSrc) {
+		return imgSrc, nil
+	}
+
+	imgDescriptor, err := crane.Head(imgSrc)
+	if err != nil {
+		return "", fmt.Errorf("unable to pull image (%s) to get the sha256 digest: %w", imgSrc, err)
```

Comment:
```suggestion
		return "", err
```
IMO it would be better to just return err here, since you're already giving an error message in prepare.go it ends up repeating itself currently
```
Problem getting image with sha for ghcr.io/stefanprodan/podinfo:6.14.0: unable to pull
          image (ghcr.io/stefanprodan/podinfo:6.14.0) to get the sha256 digest: HEAD
          https://ghcr.io/v2/stefanprodan/podinfo/manifests/6.14.0: unexpected status code 404 Not
          Found (HEAD responses have no body, use GET for details)
```